app/Http/Controllers/PrintController.php

```diff
@@ -37,4 +69,46 @@ public function modify_balance(Request $request) {
         }
         return redirect()->route('print');
     }
-}
+
+    private function print_file($file, $cost, $two_sided, $number_of_copies) {
+        $printer_name = config('app.printer_name');
+        $state = "QUEUED";
+        try {
+            $path = $file->storeAs('', md5(rand(0, 100000) . date('c')) . '.pdf', 'printing');
+            $path = Storage::disk('printing')->getDriver()->getAdapter()->applyPathPrefix($path);
+            $result = exec("lp -d " . $printer_name
+                        . ($two_sided ? " -o sides=two-sided-long-edge " : " ") 
+                        . "-n " . $number_of_copies
+                        . $path . " 2>&1");
+            if (!preg_match("/^request id is ([^\s]*) \\(1 file\\(s\\)\\)$/", $result, $job_id)) {
+                Log::error("Printing error at line: ".__FILE__.":".__LINE__." (in function ".__FUNCTION__."). result:"
+                    . print_r($result, true));
+                $state = "ERROR";
+            }
+            $job_id = $job_id[1];
+        }
+        catch (\Exception $e){
```

Comment:
What does the backslash do here?

Reply:
Backslash is the namespace separator. A `\` before the beginning of a function represents the Global Namespace.